Source/WebCore/dom/Node.cpp

```diff
@@ -2041,7 +2041,7 @@ inline void Node::moveShadowTreeToNewDocument(ShadowRoot& shadowRoot, Document&
         node.moveNodeToNewDocument(oldDocument, newDocument);
     }, [&oldDocument, &newDocument](ShadowRoot& innerShadowRoot) {
         RELEASE_ASSERT_WITH_SECURITY_IMPLICATION(&innerShadowRoot.document() == &oldDocument);
-        innerShadowRoot.moveShadowRootToNewDocument(newDocument);
+        innerShadowRoot.moveShadowRootToNewDocument(oldDocument, newDocument);
```

Comment:
This new parameter shouldn't be needed. Per assertion above you can just read ShadowRoot::document().

Reply:
There are actually 2 call sites. For this one, I would agree. The issue is with the second call site:
```
void ShadowRoot::moveShadowRootToNewParentScope(TreeScope& newScope, Document& newDocument)
{
    auto& oldDocument = documentScope();
    setParentTreeScope(newScope);
    moveShadowRootToNewDocument(oldDocument, newDocument);
}
```

`setParentTreeScope(newScope)` updates the shadowRoot's documentScope to be newDocument, *before* moveShadowRootToNewDocument() gets called.